artemis-server/src/main/java/org/apache/activemq/artemis/spi/core/remoting/Acceptor.java

```diff
@@ -83,4 +83,15 @@
    default ProtocolHandler getProtocolHandler() {
       return null;
    }
+
+   /**
+    * This is a utility method for Socket-based acceptor implementations to get the actual port used.
+    * This is useful for configurations which specify a port number of 0 which allows the JVM to select
+    * an ephemeral port.
+    *
+    * @return the actual port used if using a Socket-based acceptor implementation; null otherwise
+    */
+   default Integer getActualPort() {
+      return null;
```

Comment:
Perhaps just -1 rather than a null? Keeps it all-int, avoiding need for Integer boxing the actual impl doesnt need and potential for NPE, while still being equally invalid as a port number?

Reply:
Done.